tests/gtest/aviftilingtest.cc

```diff
@@ -0,0 +1,112 @@
+// Copyright 2022 Google LLC. All rights reserved.
+// SPDX-License-Identifier: BSD-2-Clause
+
+#include "avif/avif.h"
+#include "avif/internal.h"
+#include "gtest/gtest.h"
+
+namespace {
+
+TEST(TilingTest, SetTileConfiguration) {
+  constexpr int kThreads = 8;
+  int tile_rows_log2;
+  int tile_cols_log2;
+  // 144p
+  avifSetTileConfiguration(kThreads, 256, 144, &tile_rows_log2,
+                           &tile_cols_log2);
+  EXPECT_EQ(tile_rows_log2, 0);
+  EXPECT_EQ(tile_cols_log2, 0);
+  avifSetTileConfiguration(kThreads, 144, 256, &tile_rows_log2,
+                           &tile_cols_log2);
+  EXPECT_EQ(tile_rows_log2, 0);
+  EXPECT_EQ(tile_cols_log2, 0);
```

Comment:
Optional: To reduce this test's length, consider having a separate symmetry test instead of doing everything twice:

```cpp
TEST(TilingTest, Symmetry) {
  for (uint32_t width : {1, 144, 256, ...}) {
    for (uint32_t height : {1, 144, 256, ...}) {
      EXPECT_EQ(GetTileConfig(kThreads, width, height), GetTileConfig(kThreads, height, width));
    }
  }
}
```
It could also be integrated directly into the helper function suggested above.